lib/build_data.sh

```diff
@@ -134,17 +134,19 @@ function build_data::get_previous() {
 	fi
 }
 
-# Returns the current time in milliseconds since the UNIX Epoch.
+# Returns the current time since the UNIX Epoch, as a float with microseconds precision
 #
 # Usage:
 # ```
-# local dependencies_install_start_time=$(build_data::current_unix_time_ms)
+# local dependencies_install_start_time=$(build_data::current_unix_realtime)
 # # ... some operation ...
 # build_data::set_duration "dependencies_install_duration" "${dependencies_install_start_time}"
 # ```
-function build_data::current_unix_time_ms() {
-	date +%s%3N
-}
+function build_data::current_unix_realtime() (
+	# we use a subshell with LC_ALL=C to ensure output format is not affected by system locale
+	LC_ALL=C
+	echo "${EPOCHREALTIME}"
+)
```

Comment:
(Just since the subshell was too hard to spot before, since the brackets looked like braces, it took me several times of looking at it to see them)

```suggestion
function build_data::current_unix_realtime() {
	# We use a subshell with `LC_ALL=C` to ensure the output format isn't affected by system locale.
	(
		LC_ALL=C
		echo "${EPOCHREALTIME}"
	)
}
```